### DISCUSSION TITLE
Stuck with DNS.yml testing

### DISCUSSION THREAD

**Discussion by lifeboy** [acmesh-official/acme.sh#5274] [General]
I'm really struggling to come to grips with the automated testing in Github.  I may have finally figured out how to set secrets so the script will run, but then again I don't know.  I created a TokenName1 TokenValue1 pair for each environment variable or parameter set in the accountconf.conf file, but don't see how the script will read these.  When I run my changed code from the command like, I use something like this: ```./acme.sh  --issue  --domain 'fw.fast.za.net' --dns 'dns_miab' --home '/tmp/acme/Primary/' --accountconf '/home/roland/accountconf.conf' ``` and it works.  So how do set the secrets up so the DNS.yml script test reads the account details from the secrets instead of the config file?

The script fails because of this:
```
##[debug]Result: 'echo "Read this:   https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Test"
##[debug]if [ "lifeboy" != "acmesh-official" ]; then
##[debug]  false
##[debug]fi
##[debug]'
##[debug]Loading env
Run echo "Read this:   https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Test"
```

Of course ```"lifeboy" != "acmesh-official"``` is true, since I'm running this in my forked project, but what is the point of it?

**Comment by vmmello** [comment]
Hi,

I see on your repository that you already found it out. I'm just replying because I got this same problem and I didn't find it answered elsewhere. So answering may help others.

This error means that the required repository variables **TokenName1** and/or **TokenValue1** are not set (the error msg like this `"lifeboy" != "acmesh-official"` is a bit misleading because it seems to indicate that the user is testing in the wrong place, though it's just an internal conditional to display or not the help msg).

The problem was that you created **environment variables** not **repository variables** (that is actually the correct thing to do). Github changed the UI and the instructions on the wiki page are not precise anymore as now there are two types of variables that can be created.

Anyway, the solution is to create **TokenName1** and **TokenValue1** as **repository variables** (not as environment variables, that is the first option that Github shows in the page).
